astropy/io/misc/asdf/tags/transform/basic.py

```diff
@@ -28,8 +28,8 @@ def _from_tree_base_transform_members(cls, model, node, ctx):
         # TODO: Remove domain in a later version.
         if 'domain' in node:
             model.bounding_box = cls._domain_to_bounding_box(node['domain'])
-        elif 'bounding_box' in node:
-            model.bounding_box = node['bounding_box']
+        if 'bounding_box' in node:
```

Comment:
Why the change from ``elif`` to ``if``? If you want the 'bounding_box' clause to take precedence, maybe put it first and then use an ``elif``?

Reply:
I was considering removing `domain`. It was replaced by `bounding_box` quite a while ago.
@drdavella Is there any reason NOT to remove `domain`? I believe it was only used by jwst and by now there shouldn't be any files left with it.

Reply:
@nden it's okay with me to remove. It was removed in v1.1.0 of the Standard, so there's no requirement to support backwards compatibility here given the version of `TransformType` is already pinned to 1.1.0.

Reply:
After more conversations locally - removing it now doesn't appear to be a problem.